scrapers/MissaX.py

```diff
@@ -0,0 +1,132 @@
+    studio = STUDIO_MAP[studio] # studio map
+    log.trace(f'Studio:{studio}')
+    performers = tree.xpath('//p[@class="dvd-scenes__data" and contains(text(), "Featuring:")]//a/text()') #performers scrape
+    log.trace(f'Performers:{performers}')
+    tags = tree.xpath('//p[@class="dvd-scenes__data" and contains(text(), "Categories:")]//a/text()') #tags scrape
+    log.trace(f'Tags:{tags}')
+    details = tree.xpath('//p[count(preceding-sibling::p[@class="dvd-scenes__title"])=1]/text()|//p[@class="text text--marg"]/strong/text()|//p/em/text()') #details scrape
+    details = ''.join(details) #join details
+    details = '\n'.join(' '.join(line.split()) for line in details.split('\n')) #get rid of double spaces
+    details = re.sub("\r?\n\n?", r'\n', details) #get rid of double newlines
+    log.trace(f'Details:{details}')
+    bad_cover_url = tree.xpath("//img[@src0_4x]/@src0_4x") #cover from scene's page if better one is not found (it will be)
+    datauri = "data:image/jpeg;base64,"
+    b64img = scrape_cover(scraper, studio, title, bad_cover_url)
+    return output_json(title,tags,date,details,datauri,b64img,studio,performers)
+
+
+def scrape_cover(scraper, studio, title, bad_cover_url):
+    p = 1
+    # loop throught search result pages until img found
+    while p<20:
+        log.trace(f'Searching page {p} for cover')
+        url = 'https://'+studio.replace(" ", "")+'.com/tour/search.php?query='+urllib.parse.quote(title)+f'&page={p}'
+        tree = scraped_content(scraper, url) #get page content
+        tree = html.fromstring(tree) #parse html
+        if tree.xpath('//*[@class="photo-thumb video-thumb"]'): #if any search results present
+            try:
+                imgurl = tree.xpath(f'//img[@alt="{title}"]/@src0_4x')[0]
+                img = scraped_content(scraper, imgurl)
+                b64img = base64.b64encode(img)
+                log.trace('Cover found!')
+                return b64img
+            except:
+                if tree.xpath('//li[@class="active"]/following-sibling::li'): #if there is a next page
+                    p+=1
+                else:
+                    break
+        else:
+            break
+    #just a failsafe
+    log.warning('better cover not found, returning the bad one')
+    img = scraped_content(scraper, bad_cover_url)
+    b64img = base64.b64encode(img)
+    return b64img
+    
+
+
+def output_json(title,tags,date,details,datauri,b64img,studio,performers):
+    return {
+    'title': title,
+    'tags': [{
+        'name': x
+    } for x in tags],
+    'date': date,
+    'details': details.strip(),
+    'image': datauri + b64img.decode('utf-8'),
+    'studio': {
+        'name': studio
+    },
+    'performers': [{
+        'name': x.strip()
+    } for x in performers]
+}
+
+
+
+# frag = {"url": "https://allherluv.com/tour/trailers/Like-I-Do.html"}
+
+frag = json.loads(sys.stdin.read())
+if not frag['url']:
+    log('No URL entered.')
```

Comment:
Should be ?
```py
    log.error('No URL entered.')
    sys.exit()
```
( a url should always be present since it is a byURL only scraper, it doesnt hurt to check though)